actioncable/app/javascript/action_cable/connection.js

```diff
@@ -134,7 +141,12 @@ Connection.prototype.events = {
         return this.monitor.recordPing()
       case message_types.confirmation:
         this.subscriptions.confirmSubscription(identifier)
-        return this.subscriptions.notify(identifier, "connected")
+        if (this.reconnectAttempted) {
+          this.reconnectAttempted = false
+          return this.subscriptions.notify(identifier, "connected", {reconnected: true})
+        } else {
+          return this.subscriptions.notify(identifier, "connected")
+        }
```

Comment:
```suggestion
        const result = this.subscriptions.notify(identifier, "connected", {reconnected: this.reconnectAttempted});
        this.reconnectAttempted = false
        return result
```

This way you can check `arg.reconnected` without first checking if it's undefined.

Reply:
Actually, cleaner:

```js
const reconnected = this.reconnectAttempted
this.reconnectAttempted = false
return this.subscriptions.notify(identifier, "connected", {reconnected: reconnected})
```

Reply:
After that I'm good to merge.

Reply:
Or even better by renaming `triedToReconnect()` to `reconnectAttempted()`:
```js
const reconnected = this.reconnectAttempted()
return this.subscriptions.notify(identifier, "connected", {reconnected: reconnected})
```

Reply:
It seems like the linter doesn't allow me to write declarations inside case statements. So, it will be a one-liner.
```js
return this.subscriptions.notify(identifier, "connected", {reconnected: this.reconnectedAttempted()})
```

Reply:
The PR is updated.